module3/lessons/intro-to-cypress-prework.md

```diff
@@ -0,0 +1,437 @@
+However, since post is happening when the user clicks the submit, we don't need to have that intercept in the beforeEach block, since that doesn't need to happen for every test. But instead of adding it in after click we want to add it at the beginning of the test. 
+```js
+it('adds a new idea to the list', () => {
+    cy.intercept("POST", "http://localhost:3001/ideas", {
+      statusCode: 201,
+      body: {
+        title: "newIdeaTitle",
+        description: "newIdeaDescription"
+      }
+    }) 
+    cy.get('form input[name="title"]').type('newIdeaTitle')
+    cy.get('form input[name="description"]').type('newIdeaDescription')
+    cy.get('form button').click()
+})     
+```
+Note that we add the `cy.intercept` at the beginning of the test block, before any interactions. This ensures that Cypress is ready to intercept the POST request as soon as it's made. If we placed it after the click, we might miss intercepting the request.
+
+```js
+it('adds a new idea to the list', () => {
+    cy.intercept("POST", "http://localhost:3001/ideas", {
+      statusCode: 201,
+      body: {
+        title: "newIdeaTitle",
+        description: "newIdeaDescription"
+      }
+    }) 
+    cy.get('form input[name="title"]').type('newIdeaTitle')
+    cy.get('form input[name="description"]').type('newIdeaDescription')
+    cy.get('form button').click()
+    cy.get('.ideas-container > :nth-child(4)').should('contain', 'Test new idea')
+    cy.get('.ideas-container > :nth-child(4)').should('contain', 'Test new idea')
+})
+```
+When testing POST requests, it's crucial to set up the interception before any user actions that trigger the request.
+This ensures that Cypress is prepared to intercept and mock the response as soon as the request is made. Here's why this approach is important:
+
+1. It allows us to control the response data, enabling consistent and predictable tests.
+2. It prevents actual network requests during testing, which can make tests faster and more reliable.
+3. It enables us to test various scenarios, including error states, without modifying the server." to "It enables us to test various scenarios, including error states, without modifying the server.
+
+Remember to place your `cy.intercept()` call at the beginning of your test (`it` block), before any interactions that might trigger the POST request.
+
+After clicking the submit button, we need to add more assertions to verify that the new idea has been successfully added to the list. This is crucial because:
+
+1. It confirms that the UI has been updated correctly after the POST request.
+2. It ensures that the user can see the newly added idea, which is a key part of the user experience.
+3. It verifies that our application is correctly handling the response from the intercepted POST request.
+
+Let's add these assertions:
+
+
+
```

Comment:
It looks like some code might be missing here.   Are you still working on this part?

Reply:
Thanks @hfaerber - This has been resolved [here](https://github.com/turingschool/curriculum-site/pull/274/commits/26b6488b9202f4f9773c5696260047fc147fe428)